src/index.ts

```diff
@@ -0,0 +1,77 @@
+import Eth from 'ethjs';
+import registryMap from './registry-map.json';
+import abi from './abi.json';
+
+interface HttpProvider {
+  host: string;
+  timeout: number;
+}
+
+interface Provider {
+  network: string;
+  provider: HttpProvider;
+}
+
+interface DeployedRegistryContract {
+  entries (bytes: Uint8Array): unknown[];
+}
+
+export class MethodRegistry {
+  eth: Eth;
+
+  provider: HttpProvider;
+
+  registry: DeployedRegistryContract;
+
+  constructor(opts: Provider) {
+    if (!opts.provider) {
+      throw new Error("Missing required 'provider' option");
+    }
+    this.provider = opts.provider;
+    this.eth = new Eth(this.provider);
+    const address = (registryMap as Record<string, string>)[opts.network || '1'];
+
+    if (!address) {
+      throw new Error('No method registry found on the requested network.');
+    }
+
+    this.registry = this.eth.contract(abi as any).at(address);
+  }
+
+  async lookup(bytes: any) {
+    const result = await this.registry.entries(bytes);
```

Comment:
Oh, I was wrong about the type of `bytes` in my suggestion for the declarations file. Per the readme, the type should be `string`, and we should add a docstring explaining that it must be a `0x`-prefixed hexadecimal number. Something like:
```javascript
/**
 * @param bytes - The `0x`-prefixed hexadecimal string representing the four-byte signature of the contract method to lookup.
 */
```

Reply:
In addition, the return type of `registry.entries` should be changed from `unknown[]` to `string[]` (see usage in extension [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/helpers/utils/transactions.util.js/#L77-L88)), and the return type of `lookup` should be changed to `string | undefined` (TypeScript will probably automatically infer this once you change the return type of `registry.entries`).